test-snapshots/snapshots-output/Snapshot.ConstructorAccessor.ss

```diff
@@ -0,0 +1,97 @@
+#!r6rs
+#!chezscheme
+(library
+  (Snapshot.ConstructorAccessor lib)
+  (export
+    First*
+    First-value0
+    First?
```

Comment:
At first, I thought the uncurried version of `First*` wasn't being exported here. Then I realized that `First` only takes a single argument. 

In cases like this (a ctor that takes 1 argument), should the constructor name be `First`? 

We could always export two constructors (a curried and uncurried version), but the only reason why that might be desirable is if one wanted to use the uncurried constructor directly in FFI files, which I think would be a bad practice to encourage.

Reply:
Fixed in [217e8dc](https://github.com/purescm/purescm/pull/85/commits/217e8dc6c1ebddbfd06b9fb87e691a6e4b534075), but I'm not very happy with the overall state of the code. I feel like I'm spreading too much stuff across `codegenTopLevelBinding'`, `printDefinition` and `definitionIdentifiers`.